app/src/main/java/com/epfl/beatlink/viewmodel/library/PlaylistViewModel.kt

```diff
@@ -41,11 +41,38 @@ class PlaylistViewModel(private val repository: PlaylistRepository) : ViewModel(
 
   // Initialization block that runs when ViewModel is created
   init {
-    repository.init(onSuccess = { getPlaylists() })
+    repository.init(onSuccess = { fetchData() })
+  }
+
+  fun getUserId(): String? {
+    return repository.getUserId()
+  }
+
+  fun fetchData() {
+    getPlaylists()
+    getSharedPlaylists()
+    getPublicPlaylists()
   }
 
   fun getPlaylists() {
-    repository.getPlaylists(onSuccess = { playlistList_.value = it }, onFailure = {})
```

Comment:
Again the list of the user's playlist can have a more explicit name to better understand that he is the owner of these.